commons-component-common/src/test/java/org/exoplatform/services/thumbnail/ImageResizeServiceImplTest.java

```diff
@@ -0,0 +1,75 @@
+package org.exoplatform.services.thumbnail;
+
+import org.junit.Before;
+import org.junit.Test;
+
+import javax.imageio.ImageIO;
+import java.awt.image.BufferedImage;
+import java.io.ByteArrayInputStream;
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+
+import static org.junit.Assert.*;
+
+public class ImageResizeServiceImplTest {
+
+  private ImageResizeService imageResizeService;
+
+  private byte[]             testImageContent;
+
+  @Before
+  public void setUp() throws Exception {
+    this.imageResizeService = new ImageResizeServiceImpl();
+    File file = new File(getClass().getClassLoader().getResource("images/testresize.jpg").getFile());
+    testImageContent = Files.readAllBytes(file.toPath());
+  }
+
+  @Test
+  public void scaleImage() throws Exception {
+    byte[] resizedImage = imageResizeService.scaleImage(testImageContent, 0, 0, false, false);
+    BufferedImage bufferedImage = toBufferedImage(resizedImage);
+    assertTrue(resizedImage.length > 0);
+    assertEquals(427, bufferedImage.getHeight());
+    assertEquals(640, bufferedImage.getWidth());
+
+    resizedImage = imageResizeService.scaleImage(testImageContent, 15, 15, false, false);
+    assertTrue(resizedImage.length > 0);
+    bufferedImage = toBufferedImage(resizedImage);
+    assertEquals(15, bufferedImage.getHeight());
```

Comment:
shoud we add an assertNotEquals for width ?

Reply:
Your proposition is correct for this test case, but in reality we can't guess it, it may be equals to 15 if the width and height were equals with applying an automatic fit.
But Ok

Reply:
For the image used in your test, width and height are not equals, right ? So we know that widht is not 15 after the scale